akka-stream/src/main/scala/akka/stream/impl/io/InputStreamSinkStage.scala

```diff
@@ -25,7 +25,9 @@ private[stream] object InputStreamSinkStage {
   case object Close extends AdapterToStageMessage
 
   sealed trait StreamToAdapterMessage
-  case class Data(data: ByteString) extends StreamToAdapterMessage
+  case class Data(data: ByteString) extends StreamToAdapterMessage {
+    require(data.nonEmpty)
+  }
```

Comment:
not sure we need this check -- it has runtime cost after all, and we always avoid creating an empty one in any case now it seems anyway already on line 80?

Reply:
It's just for readability. It might be enough to comment.

Reply:
Yes, please change to that :)